sdk/appconfiguration/app-configuration/src/featureFlag.ts

```diff
@@ -52,318 +42,90 @@ export interface FeatureFlagParam extends ConfigurationSettingParam {
   enabled: boolean;
 }
 
-/**
- * FeatureFlag represents a configuration setting that stores a feature flag value.
- *
- * [More Info](https://docs.microsoft.com/azure/azure-app-configuration/concept-feature-management)
- */
-export interface FeatureFlag extends FeatureFlagParam, ConfigurationSetting {}
-
-/**
- * Targeting Client filter for the feature flag configuration setting.
- *
- * [More Info](https://docs.microsoft.com/en-us/azure/azure-app-configuration/howto-feature-filters-aspnet-core)
- */
-export interface FeatureFlagTargetingClientFilter {
+export const FeatureFlagHelper = {
   /**
-   * The name of the feature filter.
+   * Takes the ConfigurationSetting and returns the FeatureFlag.
    */
-  name: "Microsoft.Targeting";
-  /**
-   * Parameters that can be passed in the filter.
-   */
-  parameters: {
-    audience: {
-      users: string[];
-      groups: {
-        name: string;
-        rolloutPercentage: number;
-      }[];
-      defaultRolloutPercentage: number;
-    };
-  };
-}
-
-/**
- * Time window Client filter for the feature flag configuration setting.
- *
- * [More Info](https://docs.microsoft.com/en-us/azure/azure-app-configuration/howto-feature-filters-aspnet-core)
- */
-export interface FeatureFlagTimeWindowClientFilter {
-  /**
-   * The name of the feature filter.
-   */
-  name: "Microsoft.TimeWindow";
-  /**
-   * Parameters that can be passed in the filter.
-   */
-  parameters: {
-    /**
-     * Start time of the time window.
-     * Expected UTCString - Example: "Wed, 01 May 2021 13:59:59 GMT"
-     *
-     * UTCString can be obtained from `new Date().toUTCString()`.
-     * Use `Date.parse()` to parse the UTCString as Date.
-     */
-    start: string;
-    /**
-     * End time of the time window.
-     * Expected UTCString - Example: "Wed, 05 May 2021 13:59:59 GMT"
-     *
-     * UTCString can be obtained from `new Date().toUTCString()`.
-     * Use `Date.parse()` to parse the UTCString as Date.
-     */
-    end: string;
-  };
-}
+  fromConfigurationSetting: (
+    setting: ConfigurationSetting
+  ): ConfigurationSetting<FeatureFlagValue> => {
+    if (!isFeatureFlag(setting)) {
+      throw new Error("Not a feature flag..");
```

Comment:
maybe some details about what setting key wasn't a valid feature flag?

```suggestion
      throw new Error("Setting ${setting.key} is not a valid feature flag");
```

Reply:
Ahh.. missed it.. I was lazy! :)